frontend/src/components/modules/Profile/ProfileGames.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from "react";
+import { Typography, Card, CardContent, CardActionArea } from "@mui/material";
+import { useNavigate } from "react-router-dom";
+import { useAuth } from "context/AuthContext";
+import api from "api/axios";
+import type { Tournament, Match, User } from "types/models";
+import { useTranslation } from "react-i18next";
+
+const ProfileGames: React.FC = () => {
+  const [tournaments, setTournaments] = useState<Tournament[]>([]);
+  const navigate = useNavigate();
+  const { userId } = useAuth();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    const fetchTournaments = async (): Promise<void> => {
+      try {
+        const tournamentsData = await api.tournaments.getAll();
+        const filteredTournaments = tournamentsData.filter((tournament) =>
+          tournament.players.some((player) => player.id === userId)
+        );
+        // Sort tournaments based on startDate
+        filteredTournaments.sort(
+          (a, b) =>
+            new Date(b.startDate).getTime() - new Date(a.startDate).getTime()
+        );
+        setTournaments(filteredTournaments);
+      } catch (error) {
+        console.error("Error fetching tournaments:", error);
+      }
+    };
+
+    void fetchTournaments();
+  }, [userId]);
+
+  // Function to get player name by ID
+  const getPlayerNameById = (players: User[], playerId: string): string => {
+    const player = players.find((player) => player.id === playerId);
+    if (player != null) {
+      return `${player.firstName} ${player.lastName}`;
+    } else {
+      return "Unknown Player";
+    }
+  };
+
+  // Function to get matches of a player in a tournament
+  const getPlayerMatches = (
+    tournament: Tournament,
+    userId: string
+  ): Match[] => {
+    return tournament.matchSchedule.filter((match) =>
+      match.players.some((player) => player.id === userId)
+    );
+  };
+
+  if (userId == null) {
+    return <div>No user ID available</div>;
+  }
+
+  return (
+    <div>
+      {/* Map through tournaments and print info */}
+      {tournaments.map((tournament, index) => (
+        <div key={index} style={{ marginBottom: "20px" }}>
+          <Typography variant="h5" sx={{ marginBottom: 4, marginTop: 4 }}>
+            {tournament.name}
+            <Typography
+              variant="subtitle1"
+              sx={{ display: "inline", marginLeft: 1 }}
+            >
```

Comment:
lisää tälle component="span", nyt tulee consoleen "Warning: validateDOMNesting(...):  h6 cannot appear as a child of h5."